tests/component/test_stream_readers.py

```diff
@@ -545,3 +547,591 @@ def test___power_binary_reader___read_many_sample_with_wrong_dtype___raises_erro
         )
 
     assert "int16" in exc_info.value.args[0]
+
+
+@pytest.fixture
+def di_single_line_task(task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device) -> nidaqmx.Task:
+    task.di_channels.add_di_chan(
+        sim_6363_device.di_lines[0].name, line_grouping=LineGrouping.CHAN_FOR_ALL_LINES
+    )
+    return task
+
+
+@pytest.fixture
+def di_single_channel_multi_line_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[:4].name),
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    return task
+
+
+@pytest.fixture
+def di_multi_channel_multi_line_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[:4].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
+    return task
+
+
+@pytest.fixture
+def di_single_channel_port_byte_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 1/2 have 8 lines each
+    task.di_channels.add_di_chan(
+        sim_6363_device.di_ports[1].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    return task
+
+
+@pytest.fixture
+def di_multi_channel_port_byte_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 1/2 have 8 lines each, but we need to skip port 0, which has 32 lines
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[32:40].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
```

Comment:
Using port reads with CHAN_PER_LINE is kind of weird, as you noticed. It's less weird to put each port in a separate channel. Then the port read returns [0xFF, 0xFF] instead of [0x01, 0x02, ..., 0x80, 0x01, 0x02, ..., 0x80].

```
task.di_channels.add_di_chan(
    sim_6363_device.di_ports[1].name,
    line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
)
task.di_channels.add_di_chan(
    sim_6363_device.di_ports[2].name,
    line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
)
```

Reply:
Holy hell this is way easier. Why did I make it so hard on myself :D